pkg/kubelet/preemption/preemption.go

```diff
@@ -92,13 +92,14 @@ func (c *CriticalPodAdmissionHandler) HandleAdmissionFailure(admitPod *v1.Pod, f
 func (c *CriticalPodAdmissionHandler) evictPodsToFreeRequests(admitPod *v1.Pod, insufficientResources admissionRequirementList) error {
 	podsToPreempt, err := getPodsToPreempt(admitPod, c.getPodsFunc(), insufficientResources)
 	if err != nil {
-		return fmt.Errorf("preemption: error finding a set of pods to preempt: %v", err)
+		return fmt.Errorf("preemption: error finding a set of pods to preempt for %s: %v", klog.KObj(admitPod), err)
 	}
 	for _, pod := range podsToPreempt {
 		// record that we are evicting the pod
+		message := fmt.Sprintf(messageTemplate, klog.KObj(pod))
 		c.recorder.Eventf(pod, v1.EventTypeWarning, events.PreemptContainer, message)
 		// this is a blocking call and should only return when the pod and its containers are killed.
-		klog.V(3).InfoS("Preempting pod to free up resources", "pod", klog.KObj(pod), "podUID", pod.UID, "insufficientResources", insufficientResources)
+		klog.V(3).InfoS("Preempting pod to free up resources", "pod", klog.KObj(pod), "podUID", pod.UID, "insufficientResources", insufficientResources, "admitPod", klog.KObj(admitPod))
```

Comment:
Would it be better to log the adminPod at the very beginning?

Reply:
Possibly, but I didn't want to add any net new log lines just enhance the current ones.